src/device.rs

```diff
@@ -101,7 +102,7 @@ impl<'a> Device<'a> {
             }
 
             Some(Device {
-                _context: self._context,
+                context: PhantomData,
```

Comment:
Why not continue to pass `self.context` to ensure that the correct lifetime is propagated to new objects? You can refer to the same commit in libusb-rs to see how this works: https://github.com/dcuddeback/libusb-rs/commit/0eaf40f4f1a50913e157cfc429cd6af00a45f094#diff-2c70df66cd094bd49f190698078ddb18R35.

Reply:
Type inference again, since there is only one reference / lifetime going in (the one of `&self`) and only one going out (the one of `Device<'a>`),  the compiler can match them up.